library/firewall_lib.py

```diff
@@ -269,6 +323,39 @@ def handle_interface_permanent(
             fw_settings.addInterface(item)
 
 
+pci_ids = None
+
+
+def get_interface_pci():
+    pci_dict = {}
+    interfaces = nm_get_interfaces()
+    for interface in interfaces:
+        # udi/device/[vendor, device]
+        interface_ids = []
+        device_udi = nm_get_client().get_device_by_iface(interface).get_udi()
+        device_path = os.path.join(device_udi, "device")
+        for field in ["vendor", "device"]:
+            with open(os.path.join(device_path, field)) as _file:
+                interface_ids.append(_file.readline().strip(" \n")[2:])
+        interface_ids = ":".join(interface_ids)
+        if interface_ids not in pci_dict:
+            pci_dict[interface_ids] = [interface]
+        else:
+            pci_dict[interface_ids].append(interface)
+    return pci_dict
+
+
+def parse_pci_id(module, item):
+    if PCI_REGEX.search(item):
+        global pci_ids
+        if not pci_ids:
+            pci_ids = get_interface_pci()
+        interface_name = pci_ids.get(item)
+        if interface_name:
+            return interface_name
+    module.fail_json(msg="No network interfaces found with PCI device ID %s" % item)
```

Comment:
TODO: error if PCI id is invalid, rather than a single fail message in parse_pci_id. Unit test for both errors